src/sqlfluff/core/parser/grammar/sequence.py

```diff
@@ -15,15 +15,54 @@
 from sqlfluff.core.parser.helpers import trim_non_code_segments, check_still_complete
 from sqlfluff.core.parser.match_result import MatchResult
 from sqlfluff.core.parser.match_wrapper import match_wrapper
+from sqlfluff.core.parser.matchable import Matchable
 from sqlfluff.core.parser.context import ParseContext
 from sqlfluff.core.parser.grammar.base import (
     BaseGrammar,
+    MatchableType,
     cached_method_for_parse_context,
 )
 from sqlfluff.core.parser.grammar.conditional import Conditional
 from os import getenv
 
 
+def _all_remaining_metas(
+    remaining_elements: Iterable[MatchableType], parse_context: ParseContext
```

Comment:
Since we iterate over it multiple times, I _think_ `remaining_elements` should be a `Sequence` instead. Iterables don't necessarily allow iterating over them more then once (e.g. generators).

Reply:
yeah I think that's a good idea